static/templates/bookend.hbs

```diff
@@ -2,7 +2,16 @@
 
 {{#if bookend_content}}
 <div class="{{#if trailing}}trailing_bookend{{/if}} bookend sub-unsub-message">
-    <span>{{bookend_content}}</span>
+    <span>
+        {{#if is_spectator }}
+            {{#tr}}
+                Browse <z-link>recent topics</z-link>.
+                {{#*inline "z-link"}}<a href="#recent_topics">{{> @partial-block}}</a>{{/inline}}
+            {{/tr}}
+        {{else}}
+            {{bookend_content}}
+        {{/if}}
```

Comment:
The way that bookend content is computed is kinda a mess, split between this template and the JS code.  That split dates to at least 568c33f2949b69fd7b3841a202d3e5a4ee3a9cab and doesn't seem super carefully thought out.

It's kinda a pain to disentangle, but I think the reasonable thing to do here would be to refactor things so that these functions get inlined into the template:
```
    subscribed_bookend_content(stream_name) {                                                            
        return $t({defaultMessage: "You subscribed to stream {stream}"}, {stream: stream_name});         
    }                                                                                                    
                                                                                                         
    unsubscribed_bookend_content(stream_name) {                                                          
        return $t({defaultMessage: "You unsubscribed from stream {stream}"}, {stream: stream_name});     
    }                                                                                                    
                                                                                                         
    not_subscribed_bookend_content(stream_name) {                                                        
        return $t(                                                                                       
            {defaultMessage: "You are not subscribed to stream {stream}"},                               
            {stream: stream_name},                                                                       
        );                                                                                               
    }                                                                                                    
                                                                                                         
    deactivated_bookend_content() {                                                                      
        return $t({defaultMessage: "This stream has been deactivated"});                                 
    }                                                                                                    
```

That said, I'm not sure whether it's worth blocking this PR on that cleanup, which would have the nice benefit that this new code doesn't end up inside the `sub-unsub-message` class.

Reply:
Okay, did the cleanup.